pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1235,6 +1238,14 @@ func TestExpectationsOnRecreate(t *testing.T) {
 		t.Fatal("queue is shutting down")
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentReplicaSetTerminatingReplicas) {
```

Comment:
Why is this condition needed, `processNextWorkItem` was already invoked earlier, and that should've handled the necessary updates to `.status.terminatingReplicas`?

Reply:
Yes, the first invocation handles the update to `.status.terminatingReplicas`, but since it changes the RS it will cause the RS to be queued one more time.

It can then fail the test couple of lines below:

https://github.com/kubernetes/kubernetes/blob/399bd1657aed253bb7ac0c136f656774e56d4895/pkg/controller/replicaset/replica_set_test.go#L1260-L1262

This check and the item insertion to the queue are racy, so it is better to do the empty sync/empty the queue at the start.